llarp/config.cpp

```diff
@@ -35,13 +35,300 @@ namespace llarp
+  f << "# uncomment these to manually set public address and port" << std::endl;
+  f << "# this is required on providers like AWS because of their firewall "
+       "rules"
+    << std::endl;
+  f << "# public-address=your.ip.goes.here" << std::endl;
+  f << "# public-port=1090" << std::endl;
+  f << std::endl;
+  f << "# number of crypto worker threads " << std::endl;
+  f << "threads=4" << std::endl;
+  f << "# path to store signed RC" << std::endl;
+  f << "contact-file=" << basepath << "self.signed" << std::endl;
+  f << "# path to store transport private key" << std::endl;
+  f << "transport-privkey=" << basepath << "transport.private" << std::endl;
+  f << "# path to store identity signing key" << std::endl;
+  f << "ident-privkey=" << basepath << "identity.private" << std::endl;
+  f << "# encryption key for onion routing" << std::endl;
+  f << "encryption-privkey=" << basepath << "encryption.private" << std::endl;
+  f << std::endl;
+  f << "# uncomment following line to set router nickname to 'lokinet'"
+    << std::endl;
+  f << "# nickname=lokinet" << std::endl;
+  f << std::endl << std::endl;
+
+  f << "# ROUTERS ONLY: publish network interfaces for handling inbound traffic"
+    << std::endl;
+  f << "[bind]" << std::endl;
+  // get ifname
+  std::string ifname;
+  if(llarp::GetBestNetIF(ifname, AF_INET))
+    f << ifname << "=1090" << std::endl;
+  else
+    f << "# could not autodetect network interface" << std::endl
+      << "# eth0=1090" << std::endl;
+
+  f << std::endl;
+}
+
+bool
+llarp_ensure_client_config(std::ofstream &f, std::string basepath)
+{
+  f << "# ROUTERS ONLY: router settings block" << std::endl;
+  f << "# router settings block" << std::endl;
+  f << "[router]" << std::endl;
+  f << "# uncomment these to manually set public address and port" << std::endl;
+  f << "# this is required on providers like AWS because of their firewall "
+       "rules"
+    << std::endl;
+  f << "#public-address=your.ip.goes.here" << std::endl;
+  f << "#public-port=1090" << std::endl;
+  f << std::endl;
+  f << "# number of crypto worker threads " << std::endl;
+  f << "#threads=4" << std::endl;
+  f << "# path to store signed RC" << std::endl;
+  f << "#contact-file=" << basepath << "self.signed" << std::endl;
+  f << "# path to store transport private key" << std::endl;
+  f << "#transport-privkey=" << basepath << "transport.private" << std::endl;
+  f << "# path to store identity signing key" << std::endl;
+  f << "#ident-privkey=" << basepath << "identity.private" << std::endl;
+  f << "# encryption key for onion routing" << std::endl;
+  f << "#encryption-privkey=" << basepath << "encryption.private" << std::endl;
+  f << std::endl;
+  f << "# uncomment following line to set router nickname to 'lokinet'"
+    << std::endl;
+  f << "#nickname=lokinet" << std::endl;
+  f << std::endl << std::endl;
+
+  f << "[bind]" << std::endl;
+  std::string ifname;
+
+  if(llarp::GetBestNetIF(ifname, AF_INET))
```

Comment:
clients shouldn't even need a `bind` section

Reply:
well I left it commented out, so if they chose to change their client into a router, they could without having to look up the reference.